baybe/utils/chemistry.py

```diff
@@ -72,158 +72,117 @@ def name_to_smiles(name: str) -> str:
-    """
-    mols = []
-    for smiles in smiles_list:
-        try:
-            mol = Chem.MolFromSmiles(smiles)
-            if mol is None:
-                raise ValueError()
-            mols.append(mol)
-        except Exception as ex:
-            raise ValueError(
-                f"The SMILES {smiles} does not seem to be chemically valid."
-            ) from ex
-    return mols
-
-
-def smiles_to_rdkit_features(
-    smiles_list: list[str], prefix: str = "", dropna: bool = True
+def smiles_to_fingerprint_features(
+    smiles: Sequence[str],
+    encoding: SubstanceEncoding,
+    prefix: str | None = None,
+    kwargs_conformer: dict | None = None,
+    kwargs_fingerprint: dict | None = None,
 ) -> pd.DataFrame:
-    """Compute RDKit chemical descriptors for a list of SMILES strings.
+    """Compute molecular fingerprints for a list of SMILES strings.
 
     Args:
-        smiles_list: List of SMILES strings.
+        smiles: Sequence of SMILES strings.
+        encoding: Encoding used to transform SMILES to fingerprints.
         prefix: Name prefix for each descriptor (e.g., nBase --> <prefix>_nBase).
-        dropna: If ``True``, drops columns that contain NaNs.
+        kwargs_conformer: kwargs for conformer generator
+        kwargs_fingerprint: kwargs for fingerprint generator
 
     Returns:
-        Dataframe containing overlapping RDKit descriptors for each SMILES string.
+        Dataframe containing fingerprints for each SMILES string.
     """
-    mols = smiles_to_molecules(smiles_list)
-
-    res = []
-    for mol in mols:
-        desc = {
-            prefix + "RDKIT_" + dname: DTypeFloatNumpy(func(mol))
-            for dname, func in Chem.Descriptors.descList
-        }
-        res.append(desc)
-
-    df = pd.DataFrame(res)
-    if dropna:
-        df = df.dropna(axis=1)
-
-    return df
-
-
-def smiles_to_fp_features(
-    smiles_list: list[str],
-    prefix: str = "",
-    dtype: type[int] | type[float] = int,
-    radius: int = 4,
-    n_bits: int = 1024,
-) -> pd.DataFrame:
-    """Compute standard Morgan molecule fingerprints for a list of SMILES strings.
-
-    Args:
-        smiles_list: List of SMILES strings.
-        prefix: Name prefix for each descriptor (e.g., nBase --> <prefix>_nBase).
-        dtype: Specifies whether fingerprints will have int or float data type.
-        radius: Radius for the Morgan fingerprint.
-        n_bits:Number of bits for the Morgan fingerprint.
-
-    Returns:
-        Dataframe containing Morgan fingerprints for each SMILES string.
-    """
-    mols = smiles_to_molecules(smiles_list)
+    kwargs_fingerprint = kwargs_fingerprint or {}
+    kwargs_conformer = kwargs_conformer or {}
+
+    if encoding is SubstanceEncoding.MORGAN_FP:
+        warnings.warn(
+            f"Substance encoding '{encoding.name}' is deprecated and will be disabled "
+            f"in a future version. Use '{SubstanceEncoding.ECFP.name}' "
```

Comment:
Is it necessary to explicitly tell that this encoding will be used or are they equivalent (same goes for the others here)?

Reply:
What exactly do you mean? For clarification: The warning tells you what you'd need to do instead to get the exact same behavior but using the "new" approach – that's how we often do it in our deprecation warnings. Or what is your question exactly?

Reply:
yes, the intent is to tell the user what they should use and not of what the algorithm will use (but both are the same anyway)

Reply:
I consider this as resolved and will close to move forward. If you disagree, @AVHopp, please reopen